src/core/namespace.c

```diff
@@ -1107,37 +1107,64 @@ static int mount_bind_sysfs(const MountEntry *m) {
         return mount_nofollow_verbose(LOG_DEBUG, "/sys", mount_entry_path(m), NULL, MS_BIND|MS_REC, NULL);
 }
 
-static int mount_private_sysfs(const MountEntry *m) {
-        const char *entry_path = mount_entry_path(ASSERT_PTR(m));
+static int mount_private_apivfs(
+                const char *fstype,
+                const char *entry_path,
+                const char *bind_source,
+                const char *opts) {
+
         int r, n;
 
-        (void) mkdir_p_label(entry_path, 0755);
+        assert(fstype);
+        assert(entry_path);
+        assert(bind_source);
 
+        (void) mkdir_p_label(entry_path, 0755);
         n = umount_recursive(entry_path, 0);
 
-        r = mount_nofollow_verbose(LOG_DEBUG, "sysfs", entry_path, "sysfs", MS_NOSUID|MS_NOEXEC|MS_NODEV, NULL);
+        r = mount_nofollow_verbose(LOG_DEBUG, fstype, entry_path, fstype, MS_NOSUID|MS_NOEXEC|MS_NODEV, opts);
+        if (r == -EINVAL && opts)
```

Comment:
this only applies to sysfs afaics, hence add `streq(fstype, sysfs)` to the conditions here?

Reply:
or alternatively add another boolean param `try_without_options` or so that is only set for the sysfs case?

Reply:
For sysfs, `opts` is always `NULL`. So, I do not think it is necessary.